src/modules/launcher/Wox.Infrastructure/Storage/JsonStorage`1.cs

```diff
@@ -130,11 +130,18 @@ public void Save()
                 try
                 {
                     string serialized = JsonSerializer.Serialize(_data, _serializerOptions);
-                    File.WriteAllText(FilePath, serialized);
+                    using var stream = new FileStream(FilePath, FileMode.Create, FileAccess.Write, FileShare.None);
```

Comment:
It's interesting that we can change options like File share. What were the options used internally by File.WriteAllText?
Do we need to use FileShare.None or can we let other operations?

Reply:
File.WriteAllText uses FileShare.Read internally. By specifying FileShare.None, we provide exclusive access to the file during the write process. In this way, we took extra precautions against the crash that occurred by preventing access to the file with another process.

Reply:
There was a race condition in code after Powertoys update. Both update and start threads are using save function. In previous PR #30187  ` lock (_saveLock)` was added. 

After using this lock in the save method, I think there will be no difference between using FileShare.None and FileShare.Read. With FileShare.None, we prevent a possible crash by providing a second check after locking.

Additionally, the read operation that occurs while writing to the file will also attempt to read an incomplete file. I think it has some risks also. Some fields may be missing or contain incorrect information. Therefore, we must allow the reading process when the file writing is completed.

Reply:
Alright, let's give it a go, then.